rfc-capi-for-kvm-capk/README.md

```diff
@@ -0,0 +1,425 @@
+- As a member of Team Rocket, I would like to have confidence in my changes to the platform through the use of reliable unit, integration, and e2e tests.
+
+
+### Rationale
+
+With the current move to full CAPI adoption in Team Firecracker and Team Celestial, Team Rocket would be the only user and therefore the only maintainer of the current components and code that would be legacy for the other teams.
+
+This would mean that we would need to put a lot of effort into understanding and owning this code fully in order to be able to fix bugs and maintain KVM installations
+in the future.
+
+Instead of being the sole maintainer of a huge amount of legacy code, Team Rocket wants to invest this time in implementing CAPI for KVM operator and maintain relative platform parity with the other GS provider teams.
+
+### Goals
+
+- Customers can upgrade existing clusters to CAPK without significant workload downtime.
+- Existing CAPI-compatible tools such as kubectl-gs, happa, GS api, and clusterctl work with CAPK clusters.
+- Team Rocket is responsible for a significantly reduced number of repos and lines of code.
+
+### Non-goals
+
+- Get CAPK accepted as a CNCF project.
+- Support mixed CAPK/CAPV clusters.
+- Support mixed management clusters (e.g. CAPK on AWS).
+- Support `MachineSet` or `MachinePool`.
+- Support CAPI <=v1alpha3.
+
+### Scope
+
+To minimize the scope, we will start with a minimal feature set and add only features that are used by the current customers. Features that are currently not used by any customer will be dropped.
+
+Current features of `kvm-operator` (with a ✅ if we decided to keep it in CAPI):
+- [ ] configure the registry: we can set an auth token, domain and mirrors (we use that for quay for example I think); make the daemon run behind a proxy
+- [ ] configure workload cluster public key for ssh
+- [x] configure calico: CALICO_IPV4POOL_CIDR env var for calico container, CNI_MT env var for calico init container, clusterCIDR for kube-proxy)
+- [x] configure DOCKER_OPT_BIP (--bip flag) for docker daemon
+- [ ] configure domain and prefix for etcd
+- [ ] apiserver flags
+	- [ ] - configure oidc
+	- [ ] --etcd-prefix
+	- [ ] --service-cluster-ip-range
+	- [ ] --secure-port
+	- [ ] --requestheader-allowed-names
+- [ ] kubelet configuration
+	- [ ] clusterDNS
+	- [ ] clusterDomain
+	- [ ] --node-labels (flag)
+- [ ] kube-proxy
+	- [ ] conntrack.maxPerCore
+- [ ] ignition
+	- [x] dns and ntp servers
+	- [x] ssh user list
+- [ ] iscsi initiator name (masters)
+- [ ] configure storage type (host path/persistent)
+- [x] master and worker node pods: cpu and memory limits,
+- [x] size of kubelet and docker volume
+- [x] automatically delete TC node pods when NotReady for too long
+- [x] dead endpoints: don't route traffic to non-ready nodes
+- [x] vertical pod autoscaler for operator pod
+
+#### Features to drop
```

Comment:
Okay I believe this is what I meant in the earlier comment. So adding comments above if they are covered by CAPI in a single list would be nice.

Reply:
I agree it's confusing. I wasn't sure how to present this information.

Reply:
I cleaned it up. Let me know what you think.